www/docs/admin/edittopic.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+include_once '../../includes/easyparliament/init.php';
+
+$this_page = 'admin_edittopic';
+
+$topics = new \MySociety\TheyWorkForYou\Topics();
+
+$PAGE->page_start();
+$PAGE->stripe_start();
+
+$slug = get_http_var('id');
+if ($slug) {
+  $topic = $topics->getTopic($slug);
+} else {
+  $topic = new \MySociety\TheyWorkForYou\Topic();
+}
+
+$action = get_http_var('action');
+switch ($action) {
+    case 'add':
+      $success = add_topic($topic);
+      break;
+    case 'update':
+      $success = update_topic($topic);
+      break;
+    case 'setimage':
+      $success = add_image($topic);
+      break;
+    case 'addcontent':
+      $success = add_content($topic);
+      break;
+    case 'deletecontent':
+      $success = delete_content($topic);
+      break;
+    case 'addpolicysets':
+      $success = add_policy_sets($topic);
+      break;
+    case 'addpolicies':
+      $success = add_policies($topic);
+      break;
+    default:
+      $success = NULL;
+}
+
+if (!is_null($success)) {
+    if ($success) {
+        $out = "<h4>Update successful</h4>";
+    } else {
+        $out = "<h4>Failed to update Topic</h4>";
+    }
+    print $out;
+}
+
+?>
+
+  <h2><?= $topic->title() ?></h2>
+    <div id="adminbody" class="topic">
+        <form action="edittopic.php" method="post">
+          <input type="hidden" name="action" value="update">
+          <input type="hidden" name="id" value="<?= $topic->slug() ?>">
+          <label for="title">Title</label> <input id="title" name="title" value="<?= $topic->title() ?>">
+
+          <label for="search_string">Search string</label> <input id="search_string" name="search_string" value="<?= $topic->search_string() ?>">
+
+          <p>
+           <input type="checkbox" value="1" id="front_page" name="front_page" <?= $topic->onFrontPage() ? 'checked' : '' ?>> <label class="inline" for="front_page">Show on Front Page</label>
+          </p>
+
+          <label for="description">Description</label>
+          <textarea id="description" rows="5" name="description"><?= $topic->description() ?></textarea>
+
+          <p>
+          <input type="submit" value="Save">
+          </p>
+        </form>
+
+        <h3>Set Image</h3>
+        <form enctype="multipart/form-data" action="edittopic.php" method="post">
```

Comment:
Is there a reason you have to set the image separately from the other fields? Guess it's rare so doesn't really matter, just seemed a bit odd.